src/BarracksServer/Network/Send.Normal.cs

```diff
@@ -180,31 +180,107 @@ public static void ZoneTraffic()
 			}
 
 			/// <summary>
-			/// Updates the state of a postbox message.
+			/// Invokes a lua function.
+			/// </summary>
+			/// <param name="conn"></param>
+			/// <param name="str"></param>
+			public static void Run(IBarracksConnection conn, string str)
+			{
+				var packet = new Packet(Op.BC_NORMAL);
+				packet.PutInt(NormalOp.Barrack.Run);
+				packet.PutLpString(str);
+
+				conn.Send(packet);
+			}
+
+			/// <summary>
+			/// Mailbox
+			/// </summary>
+			/// <param name="conn"></param>
+			public static void Mailbox(IBarracksConnection conn)
+			{
+				var packet = new Packet(Op.BC_NORMAL);
+				packet.PutInt(NormalOp.Barrack.Mailbox);
+
+				packet.Zlib(true, zpacket =>
+				{
+					var mailBox = conn.Account.Mailbox;
+					zpacket.PutByte(mailBox.HasMessages);
+					zpacket.PutInt(mailBox.ReadMessageCount);
+					zpacket.PutInt(mailBox.MessageCount);
+					zpacket.PutInt(mailBox.MessageCount);
+
+					foreach (var message in mailBox.GetMail())
+					{
+						var items = message.GetItems();
+
+						zpacket.PutLpString(message.Sender);
+						zpacket.PutLpString(message.Subject);
+						zpacket.PutLpString(message.Message);
+						zpacket.PutDate(message.StartDate); // Date Sent?
```

Comment:
I have no clue what the other dates are because they don't seem to do anything in the client. The only one in the client-side lua is the expiration/destroy date.